### PR TITLE
Generate API Documentation With Flow Types

### LINKED ISSUE #30: Build API docs with documentation.js using Flow
To cut down on typing, use [documentation.js](https://github.com/documentationjs/documentation) for JSDOC syntax as well as Flow type support. This way types won't need to be written twice and potentially get out of sync

### PR DESCRIPTION
**Pull request by thebearjew**
1. Adds [documentation.js](https://github.com/documentationjs/documentation)
2. Use Flow types and JSDOC comments to build API markdown docs
3. Changes all chat options to use `camelCase` for a more consistent JavaScript API.
4. Convert all `camelCase` option keys into `snake_case` before passing to `keybase chat api`
5. Adds `yarn docs` to the package scripts 
6. Rebased on `nathan/api-listen` to include method changes (until that branch is merged)

Closes #30

### CODE REVIEW

File: lib/types.js [hunk 1]
```diff
@@ -1,5 +1,13 @@
 // @flow
```
Comment (thebearjew) [inline lib/types.js]: Flow types were changed from `snake_case` to `camelCase` for a better JavaScript API.

Automatic translation from snake to camel is done via `lodash/snakeCase`.

File: out [hunk 2]
```diff
@@ -0,0 +1 @@
+This should appear in output
```
Comment (nathunsmitty) [inline out]: What's this file for?

File: lib/types.js [hunk 3]
```diff
@@ -28,149 +36,146 @@ export type Pagination = {|
...
-  conversation_id: string,
+  conversationId?: string,
   query: string,
-  is_regex: boolean,
-  max_messages: number,
+  isRegex: boolean,
+  maxMessages: number,
 |}
 
+/**
```
Comment (nathunsmitty) [inline lib/types.js]: Are these empty comments so documentation.js notices the types?

File: lib/types.js (hunk 1 above)
Comment (nathunsmitty) [reply lib/types.js]: This is super clever!

File: lib/bot.js [hunk 4]
```diff
@@ -119,9 +139,38 @@ class Bot {
...
+   * you can do what you want with it. For example, if you want to write a
+   * Keybase bot that talks shit at anyone who dares approach it, this is the
+   * function to use.*
+   *
+   * Specifically, it will call the `onMessage` function you provide for every
+   * message your bot receives.*
+   *
+   *  @memberof Bot
+   *
+   *  @example
```
Comment (nathunsmitty) [inline lib/bot.js]: I updated this example in #33, I can come back and update this once we merge that in.

File: README.md [hunk 5]
```diff
@@ -44,176 +49,341 @@ try {
...
 ```
 
-### Commands
+### API
 
-Anywhere a promise is returned from the bot API, you should handle error with `.cathc(err => { ... })`
+<!-- Generated by documentation.js. Update this documentation by updating the source code. -->
 
-#### `bot.init(options)`
+##### Table of Contents
```
Comment (nathunsmitty) [inline README.md]: Is there any way to bump up the headings on the ToC? It's a little weird to go from h1 and 2s to h5s and 6s once we reach this point

Comment (thebearjew) [reply README.md]: That's done automatically by the script. As best I can tell I am not able to control it. I can tweak the parameters around `documentation.yml` to see if I can reduce the header level.

File: lib/types.js (hunk 3 above)
Comment (thebearjew) [reply lib/types.js]: Yes!

File: out (hunk 2 above)
Comment (thebearjew) [reply out]: Should not have been committed. I'll fix it.

File: README.md (hunk 5 above)
Comment (thebearjew) [reply README.md]: @nathunsmitty The answer is that the table of contents and the Markdown that is output from the library is two levels deeper than the section it is mounted to (e.g. `###API` → `##### Table of Contents`)

The markdown output cannot be customized but the HTML template out can be and included in the README if necessary.

**Pull request merged**